src/chain/chain_accumulator.rs

```diff
@@ -17,30 +17,55 @@ pub const UNRESPONSIVE_THRESHOLD: usize = 48;
 /// The period (X consensued observations) after which node be considered as unresponsive.
 pub const UNRESPONSIVE_WINDOW: usize = 64;
 
-struct VoteStatus {
-    event: AccumulatingEvent,
-    unvoted: BTreeSet<PublicId>,
+#[derive(Default)]
+struct VoteStatuses {
+    tracked_events: VecDeque<AccumulatingEvent>,
+    unvoted: BTreeMap<PublicId, BTreeSet<AccumulatingEvent>>,
```

Comment:
That does not seem ideal,
Maybe could put the AccumulatingEvent in Rc to avoid cloning these events so many times?
i.e Create an Rc push it in the VecDeque and also use it in unvoted?

Reply:
good idea, will make the change